### PR TITLE
Build pipeline for main branch. 

### PR DESCRIPTION
**Pull request by nikithauc**

### CODE REVIEW

**Review by zengin — APPROVED** [review]

File: .azure-pipelines/common-templates/security-post-checks.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      APIScan: false
+      BinSkim: false
+      CodesignValidation: true
+      CredScan: true
+      FortifySCA: false
+      FxCop: false
+      ModernCop: true
+      MSRD: false
+      PoliCheck: true
+      RoslynAnalyzers: true
```
Comment (MIchaelMainer) [inline .azure-pipelines/common-templates/security-post-checks.yml:48]:
```suggestion
      RoslynAnalyzers: false
```

File: .azure-pipelines/common-templates/security-post-checks.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      SDLNativeRules: false
+      Semmle: false
+      TSLint: false
```
Comment (MIchaelMainer) [inline .azure-pipelines/common-templates/security-post-checks.yml:51]:
```suggestion
      TSLint: true
```

File: .azure-pipelines/common-templates/security-post-checks.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  - task: PublishSecurityAnalysisLogs@3
+    displayName: "Publish Security Analysis Logs"
+    inputs:
+      ArtifactName: "CodeAnalysisLogs"
+      ArtifactType: "Container"
+      AllTools: false
+      AntiMalware: false
+      APIScan: false
+      BinSkim: false
+      CodesignValidation: true
```
Comment (MIchaelMainer) [inline .azure-pipelines/common-templates/security-post-checks.yml:41]:
```suggestion
      CodesignValidation: false
```

File: .azure-pipelines/common-templates/security-post-checks.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  - task: SdtReport@1
+    displayName: "Security Analysis Report"
+    continueOnError: true
+    condition: succeededOrFailed()
+    inputs:
+      AllTools: false
+      APIScan: false
+      BinSkim: false
+      BinSkimBreakOn: "WarningAbove"
+      CodesignValidation: true
```
Comment (MIchaelMainer) [inline .azure-pipelines/common-templates/security-post-checks.yml:15]:
```suggestion
      CodesignValidation: false
```

File: .azure-pipelines/common-templates/security-post-checks.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      CodesignValidationBreakOn: "WarningAbove"
+      CredScan: true
+      FortifySCA: false
+      FxCop: false
+      ModernCop: false
+      MSRD: false
+      PoliCheck: true
+      PoliCheckBreakOn: "Severity1"
+      RoslynAnalyzers: true
```
Comment (MIchaelMainer) [inline .azure-pipelines/common-templates/security-post-checks.yml:24]:
```suggestion
      RoslynAnalyzers: false
```

**Review by MIchaelMainer — CHANGES_REQUESTED** [review]

**Review by zengin — APPROVED** [review]

File: .azure-pipelines/ci-build-production.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+steps:
+
+- script: git checkout main
+- template: ./common-templates/security-pre-checks.yml
+- task: CopyFiles@2
+  displayName: 'Copy Files to staging directory'
+  inputs:
+    SourceFolder: '$(System.DefaultWorkingDirectory)'
+    Contents: |
```
[the remark this answers is not in the document]
Comment (nikithauc) [reply .azure-pipelines/ci-build-production.yml:25]: Done 
Please check
trigger:
  branches:
    include:
      - main
  paths:
    include:
      - microsoft-graph.d.ts

**Review by MIchaelMainer — APPROVED** [review]

**Pull request merged**